rpcserver.go

```diff
@@ -875,6 +895,15 @@ func addrPairsToOutputs(addrPairs map[string]int64) ([]*wire.TxOut, error) {
 	return outputs, nil
 }
 
+// allowCORS wraps the given http.Handler with a function that adds the
+// Access-Control-Allow-Origin header to the response.
+func allowCORS(handler http.Handler, origin string) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		w.Header().Set("Access-Control-Allow-Origin", origin)
```

Comment:
Hey @guggero, I'm coming in from https://github.com/lightningnetwork/lnd/pull/3281. 

Just wanted to ask about your use case for CORS and whether or not this will suffice. I had started with this in my PR but ended up having to expand on it with preflight request options once I deployed something on a live domain. Simple requests don't need to worry about preflight requests but if you're passing in certain headers, for instance `grpc-metadata-macaroon`, or using PUT/DELETE requests, I think you might find that you have to handle preflights: https://github.com/lightningnetwork/lnd/pull/3281/files#diff-01ada5680aaa18fc79cbf78b2d55a37bR725

It's been awhile and I stopped doing it, just providing my experience when I first went through this from soup to nuts.

Reply:
Hey Anthony. Thanks for the input. You are correct, this might not be enough. I'll add it to the TODO list.

Reply:
Thanks a lot @guggero! Great work on this, very happy to see it get introduced.